src/components/CardList.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, useState } from 'react';
+
+import classes from './CardList.module.css';
+import Card from './Card';
+import Draggable from './Draggable';
+import { CardListItemProps, CardListProps } from './CardList.types';
+import Modal from './Modal';
+import ItemDetails from './ItemDetails';
+import '../utils/array';
+
+const CardList: FC<CardListProps> = ({ cards, onUpdated }): React.ReactElement => {
+  const [dragginItemId, setDragginItemId] = useState('');
+  const [dragOver, setDragOver] = useState('');
+
+  const handleDragStart = (e: React.DragEvent<HTMLElement>): void => {
```

Comment:
Here we go! Please read https://react.dev/reference/react/useCallback I believe that's one of a good use cases (all of the callbacks here).